proxy/src/main/java/com/wavefront/agent/GraphiteStringHandler.java

```diff
@@ -41,13 +55,28 @@ public GraphiteStringHandler(final ForceQueueEnabledAgentAPI agentAPI,
                                final long millisecondsPerBatch,
                                final int pointsPerBatch,
                                final int blockedPointsPerBatch,
-                               final Formatter formatter) {
-    this.pointHandler = new PointHandler(agentAPI, daemonId, port, logLevel, validationLevel, millisecondsPerBatch,
-        pointsPerBatch, blockedPointsPerBatch);
+                               final Formatter formatter,
+                               @Nullable final String pointLineWhiteListRegex,
+                               @Nullable final String pointLineBlackListRegex) {
+    this.pointHandler = new PointHandler(agentAPI, daemonId, port, logLevel, validationLevel,
+        millisecondsPerBatch, pointsPerBatch, blockedPointsPerBatch);
 
     this.prefix = prefix;
     this.blockedPointsPerBatch = blockedPointsPerBatch;
     this.formatter = formatter;
+
+    this.pointLineWhiteList = StringUtils.isBlank(pointLineWhiteListRegex) ?
+        null : Pattern.compile(pointLineWhiteListRegex);
+
+    this.pointLineBlackList = StringUtils.isBlank(pointLineBlackListRegex) ?
+        null : Pattern.compile(pointLineBlackListRegex);
+
+    this.regexTimer = Metrics.newTimer(
+        new MetricName("validationRegex." + String.valueOf(port), "", "microsecs"),
```

Comment:
I think this is just in case there's something ridiculous (Sam was a bit worried that we couldn't predict the compute expense of the regex that the customer supplies)

(But probably the metric machinery might be as expensive as the regex =P, so I could take it out if you want )